mindspore/ccsrc/plugin/device/gpu/kernel/cuda_impl/cuda_ops/matrix_transpose_impl.cu

```diff
@@ -24,6 +24,9 @@ using Complex = mindspore::utils::Complex<T>;
 
 template <typename T>
 __global__ void MatrixTransposeKernel(const T *input, int elements, int row, int col, T *output) {
+  if (col <= 0 || row <= 0 || row != col) {
```

Comment:
Thank you for your code committion！Could you explain why the case of non-square matrix is skipped here? And have you really encountered a situation that the col or row is less or equal than 0? If so, this should be intercepted earlier on the host side.

Reply:
Sorry about that, the constraint that the matrix is square was mistakenly added because we were looking at an old version of the kernel. I've fixed the constraint to ensure that both row/col are positive numbers.